waspc/src/Wasp/Project.hs

```diff
@@ -31,24 +31,29 @@ compile ::
 compile waspDir outDir options = do
   compileWarningsAndErrors <-
     analyzeWaspProject waspDir options >>= \case
-      Left analyzerErrors -> return ([], analyzerErrors)
-      Right appSpec -> generateCode appSpec outDir options
+      (Left analyzerErrors, analyzerWarnings) -> return (analyzerWarnings, analyzerErrors)
+      (Right appSpec, analyzerWarnings) ->
+        first (<> analyzerWarnings) <$> generateCode appSpec outDir options
```

Comment:
@infomiho it generates the code, and then to the first member of the pair it returns, it adds analyzer warnings to. That is how I would read it.

@sodic you made a mistake there, you added analyzer warnings to the second member of the pair, it should have been added to the first member.
I wonder why you can't read the shorter expression? Is the `first` a function that you don't know? That one is not the most basic one, but it has quite clear name and it quite popular I think, so I think it is ok to learn it. `<>` you know, sections you know, and `<$>` also, so I am curious what is confusing.

Reply:
Don't have time to go into this now :smile: 

I added it to my discussions todo list, the current code can stay.

Reply:
I still find this line super confusing since I'm not using `first` on a daily basis. I'll get confused in a few months again after I forget what the function does.

On the other hand, the code that @sodic wrote seems basic enough.

Reply:
Sounds like the same discussion we had with `fmap`, so I will thread carefully :D.

For the record, `first` is something we have used through our codebase for a long time, and we use it in at least 13 other places in the codebase. So it is not like I am introducing something new.

Also, `first` is really simple and useful, and I would recommend making it a part of your Haskell vocabulary, it is not something esoteric!

In general, I am open to keep having these discussions if you both @infomiho @sodic  find some part of our Haskell code too weird / complex, but I think you should also keep the open mind to learning new stuff, especially when it is simple helper functions / operators that are commonly used. This is all still very much Boring Haskell, it is not like I am introducing Lenses or Prisms or an alternate effect system to Monads / Monad transformers.
Otherwise, we would have to define some kind of line on which operators we can use and which not, and that just doesn't make much sense. So let's use what we like and let's be open to learning and gaining more experience.